sympy/core/expr.py

```diff
@@ -146,11 +146,15 @@ def _hashable_content(self):
         return self._args
 
     def __eq__(self, other):
-        try:
-            other = _sympify(other)
-            if not isinstance(other, Expr):
+        if not isinstance(other, Basic):
+            if iterable(other):
```

Comment:
looks like `dict` is not considered an `iterable`, but I think it should get a similar treatment

Reply:
Dict is Basic but not Expr so it will get reject without sympification.

Reply:
I mean a python `dict`, which is not a `Basic`, nor will it go to the `iterable` branch, so it will try to `sympify` the `dict`, potentially creating `Dict` with `None` just like it did with `tuple->Tuple` before